frontend/src/views/files/Listing.vue

```diff
@@ -383,17 +383,22 @@ export default {
   },
   watch: {
     req: function () {
-      // Reset the show value
-      this.showLimit = 50;
+      if (window.sessionStorage.getItem('listFrozen') !=="true"){
+        // Reset the show value
+        this.showLimit = 50;
 
-      // Ensures that the listing is displayed
-      Vue.nextTick(() => {
-        // How much every listing item affects the window height
-        this.setItemWeight();
+        // Ensures that the listing is displayed
+        Vue.nextTick(() => {
+          // How much every listing item affects the window height
+          this.setItemWeight();
 
-        // Fill and fit the window with listing items
-        this.fillWindow(true);
-      });
+          // Fill and fit the window with listing items
+          this.fillWindow(true);
+        });
+      }
+      if (this.req.isDir) {
+        window.sessionStorage.setItem('listFrozen', "false"); 
+      }
```

Comment:
Without this modification, it still works. But we can see the listing page blinking when we are back from the preview. It's because we are keeping the listing page alive all the time, it's watching the req to refill the window even if it's in the backgound. so we need this modification here. and I use  window.sessionStorage.setItem to transfer data between .vue files. It's very fast and simple.